pkg/controller/certificaterequests/vault/vault.go

```diff
@@ -87,12 +88,16 @@ func (v *Vault) Sign(ctx context.Context, cr *v1.CertificateRequest, issuerObj v
 		return nil, nil
 	}
 
-	// TODO: distinguish between network errors and other which might warrant a failure.
 	if err != nil {
 		message := "Failed to initialise vault client for signing"
 		v.reporter.Pending(cr, err, "VaultInitError", message)
 		log.Error(err, message)
-		return nil, err // Return error to requeue and retry
+
+		if cmerrors.IsInvalidData(err) {
+			return nil, nil // Don't retry, wait for the issuer to be updated
+		} else {
+			return nil, err // Return error to requeue and retry
+		}
```

Comment:
```suggestion
		}
		return nil, err // Return error to requeue and retry
```

No need to the else here IMO.